crates/types/src/joints/leg.rs

```diff
@@ -0,0 +1,201 @@
+use std::{
+    ops::{Add, Div, Index, IndexMut, Mul, Sub},
+    time::Duration,
+};
+
+use serde::{Deserialize, Serialize};
+use serialize_hierarchy::SerializeHierarchy;
+
+#[derive(Clone, Copy)]
+pub enum LegJointsName {
```

Comment:
Naming suggestion: `LegJoint`

This applies to all other `*JointsName`

Reply:
Why singular? I would argue that e.g. `LegJoints` contains all `Joints` in the leg, so plural fits better